paddle/gserver/layers/MultinomialSampler.cpp

```diff
@@ -49,26 +50,19 @@ MultinomialSampler::MultinomialSampler(const real* prob, int size)
   int smallPos = nextSmallPos(0);
   int bigPos = nextBigPos(0);
 
-  auto fillIntervals = [&]() {
-    while (bigPos < size && smallPos < size) {
-      while (intervals_[bigPos].thresh > 1 && smallPos < size) {
-        intervals_[smallPos].otherId = bigPos;
-        intervals_[bigPos].thresh -= 1 - intervals_[smallPos].thresh;
-        smallPos = nextSmallPos(smallPos + 1);
-      }
-      bigPos = nextBigPos(bigPos + 1);
```

Comment:
Here, may shift the bigPos when intervals_[bigPos].thresh > 1 and smallPos >= size.